### PR TITLE
classpath prepending

### PR DESCRIPTION
**Pull request by fommil**

### CODE REVIEW

**Review by dwijnand — APPROVED** [review]
I'm happy to accept this. requesting review by @eed3si9n.

File: launcher-implementation/src/main/scala/xsbt/boot/ModuleDefinition.scala [hunk 1]
```diff
@@ -17,9 +17,10 @@ final class RetrievedModule(val fresh: Boolean, val definition: ModuleDefinition
   def this(fresh: Boolean, definition: ModuleDefinition, detectedScalaVersion: Option[String], baseDirectories: List[File]) =
     this(fresh, definition, detectedScalaVersion, None, baseDirectories)
 
+  lazy val monkeys: Array[File] = sys.props.get("sbt.launcher.cp.prepend").toArray.flatMap(ms => ms.split(",").map(new File(_)))
   lazy val classpath: Array[File] = getJars(baseDirectories)
-  lazy val fullClasspath: Array[File] = concat(classpath, definition.extraClasspath)
+  lazy val fullClasspath: Array[File] = concat(concat(monkeys, classpath), definition.extraClasspath)
```
Comment (eed3si9n) [inline launcher-implementation/src/main/scala/xsbt/boot/ModuleDefinition.scala:22]: Is this meant to patch the launched application like sbt, or the Scala library?

**Review by eed3si9n — CHANGES_REQUESTED** [review]
More of a question.
If this is meant to patch Scala, it might not work because we use classloader to resolve them.

**Comment by fommil** [comment]
this is not meant to patch scala or zinc, it is meant to patch the launched application. In this case, sbt.

For patching the scala compiler and zinc, see https://github.com/sbt/sbt/pull/3643

**Comment by fommil** [comment]
@eed3si9n any more requests for this? Shame we missed the 1.1.2 release.

**Review by eed3si9n — APPROVED** [review]

**Comment by eed3si9n** [comment]
LGTM

**Pull request merged**